src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDesignTimeDescriptorFactory.cs

```diff
@@ -93,6 +93,23 @@ public TagHelperAttributeDesignTimeDescriptor CreateAttributeDescriptor(Property
             return null;
         }
 
+        /// <summary>
+        /// Retrieves <paramref name="assembly"/>'s location on disk.
+        /// </summary>
+        /// <param name="assembly">The assembly.</param>
+        /// <returns>The path to the given <paramref name="assembly"/>.</returns>
+        public virtual string GetAssemblyLocation(Assembly assembly)
```

Comment:
virtual for testing. Decided to keep it public in the chance that external users need to test this as well. As it stands today it can't be unit tested for both net45/coreclr without this entry point.